demos/interactive_face_detection_demo_gapi/main.cpp

```diff
@@ -0,0 +1,599 @@
+
+    if (!no_smooth) {
+        face = matchFace(rect, prev_faces);
+        float intensity_mean = calcMean(frame(rect));
+        intensity_mean += 1.0;
+
+        if ((face == nullptr) ||
+            ((std::abs(intensity_mean - face->_intensity_mean) / face->_intensity_mean) > 0.07f)) {
+            face = std::make_shared<Face>(id++, rect);
+        } else {
+            prev_faces.remove(face);
+        }
+
+        face->_intensity_mean = intensity_mean;
+        face->_location = rect;
+    } else {
+        face = std::make_shared<Face>(id++, rect);
+    }
+}
+
+void ageGenderDataUpdate(const Face::Ptr &face,
+                         const cv::Mat &out_age,
+                         const cv::Mat &out_gender) {
+    const float *age_data =    out_age.ptr<float>();
+    const float *gender_data = out_gender.ptr<float>();
+
+    float maleProb = gender_data[1];
+    float age      = age_data[0] * 100;
+
+    face->updateGender(maleProb);
+    face->updateAge(age);
+}
+
+void headPoseDataUpdate(const Face::Ptr &face,
+                        const cv::Mat &out_y_fc,
+                        const cv::Mat &out_p_fc,
+                        const cv::Mat &out_r_fc) {
+    const float *y_data = out_y_fc.ptr<float>();
+    const float *p_data = out_p_fc.ptr<float>();
+    const float *r_data = out_r_fc.ptr<float>();
+
+    face->updateHeadPose(y_data[0], p_data[0], r_data[0]);
+}
+
+void emotionsDataUpdate(const Face::Ptr &face, const cv::Mat &out_emotion) {
+    const float *em_data = out_emotion.ptr<float>();
+
+    std::map<std::string, float> em_val_map;
+    for(size_t i = 0; i  < EMOTION_VECTOR.size(); i++) {
+        em_val_map[EMOTION_VECTOR[i]] = em_data[i];
+    }
+
+    face->updateEmotions(em_val_map);
+}
+
+void landmarksDataUpdate(const Face::Ptr &face, const cv::Mat &out_landmark) {
+    const float *lm_data = out_landmark.ptr<float>();
+
+    size_t n_lm = 70;
+
+    std::vector<float> normedLandmarks(&lm_data[0], &lm_data[n_lm]);
+
+    face->updateLandmarks(normedLandmarks);
+}
+
+bool isNumber(const std::string& str)
+{
+    if(str.empty() || ((!isdigit(str[0])) && (str[0] != '-') && (str[0] != '+')))
+        return false;
+
+   char * p;
+   strtol(str.c_str(), &p, 10);
+
+   return (*p == 0);
+}
+
+void setInput(cv::GStreamingCompiled stream, const std::string& input ) {
+    if (input == "cam") {
+        stream.setSource(cv::gapi::wip::make_src<cv::gapi::wip::GCaptureSource>(0));
+    } else if (isNumber(input)) {
+        stream.setSource(cv::gapi::wip::make_src<cv::gapi::wip::GCaptureSource>(std::stoi(input)));
+    } else {
+        stream.setSource(cv::gapi::wip::make_src<cv::gapi::wip::GCaptureSource>(input));
+    }
+}
+
+// Copied from demos/common/include/samples/common.hpp
+static std::string fileNameNoExt(const std::string &filepath) {
+    auto pos = filepath.rfind('.');
+    if (pos == std::string::npos) return filepath;
+    return filepath.substr(0, pos);
+}
```

Comment:
Is this function actually necessary? The only place it's used is to supply the `weights` parameter to `cv::gapi::ie::Params`, but `Params` has a constructor that doesn't require the `weights` parameter.

Reply:
It is necessary to invoke the constructor of `cv::gapi::ie::Params`  with `weights`
If just invoke the constructor without `weights` demo will crash in runtime

Reply:
Why does it crash? The constructor without `weights` [initializes the weights to an empty string](https://github.com/opencv/opencv/blob/4.5.1/modules/gapi/include/opencv2/gapi/infer/ie.hpp#L99), which should cause IE to try to find a weights file with the same base name as the model file, which is exactly what we want.

Reply:
Constructor with `weights`  is used to load  network
Constructor without `weights` is used to import network 
Different between **load** and **import** is that **import** - take already compiled network from blob and **load** - take ir, weights and compile network after that 

Unfortunately the logic you are talking about is not yet supported in GAPI framework
Constructor without `weights` were added for  `importNetwork`

Reply:
OK, got it.